### PR TITLE
Allow empty strings for non-required fields

### PR DESCRIPTION
**Pull request by oliverroick**
This PR addresses cadasta/cadasta-platform#592. 

GeoODK sends empty XML (`<location_problems />`) elements when no value is provided for a form field. We convert the XML into a Python dictionary using `pyxform.xform2json.XFormToDict`; all empty XML elements are converted to empty strings (`"location_problems": ""`) or a list with one empty string (`"location_problems": [""]`).

This is a problem for `integer`, `decimal`, `select_one` and `select_multiple` fields because an empty string is an actual value but it's not valid for fields mentioned above because it's not a number and usually not an option for select fields.

I'm addressing this issue by checking if the value equals either `""` or `[""]` and converting it to `None` before proceeding with validation.

### CODE REVIEW

File: tests/test_validation.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            schema=self.schema,
+            name='testattr',
+            long_name='Test attribute',
+            index=1,
+            attr_type=AttributeType.objects.get(name='select_one'),
+            choices=['a', 'b', 'c'],
+            required=True
+        )
+        with pytest.raises(ValidationError):
+            attr.validate('')
```
Comment (linzjax) [inline tests/test_validation.py:89]: Should this also check that `attr.validate([''])` fails?

File: tests/test_validation.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            name='testattr',
+            long_name='Test attribute',
+            index=1,
+            attr_type=AttributeType.objects.get(name='select_multiple'),
+            choices=['a', 'b', 'c'],
+            required=True
+        )
+
+        with pytest.raises(ValidationError):
+            attr.validate('')
```
Comment (linzjax) [inline tests/test_validation.py:119]: Same here:
Should this also check that `attr.validate([''])` fails?

**Comment by linzjax** [comment]
Oh good grief ignore me. I go this one confused with the last questionnaire upgrade that I was assigned to review. I'll see myself out...

**Review by bjohare — COMMENTED** [review]
As @linzjax  says, should the initial validation check in `models.py` be modified to cover `required` fields where the value is `['']`? (Line 284 in models.py). Other than that it looks good to me.

**Comment by oliverroick** [comment]
@linzjax @bjohare Good catch, I added the tests and changed the required check to also catch `['']` as empty values.

**Review by bjohare — APPROVED** [review]
Looks good.

**Pull request merged**